tests/framework-tests/core/js/DataBindingTests.js

```diff
@@ -1273,6 +1273,26 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         }, "source");
     });
 
+    /** FLUID-6191: Proper diagnostic on indirect model reference which fails to resolve **/
+
+    fluid.defaults("fluid.tests.fluid6191root", {
+        gradeNames: "fluid.modelComponent",
+        components: {
+            badRefHolder: {
+                type: "fluid.modelComponent",
+                options: {
+                    model: "{fluid6191root}.nonexistent.model.path"
```

Comment:
Another test case to be covered is, when the other component that the indirect model relay references to does exist but having a non-modelComponent type such as "fluid.component".

Reply:
Good catch, @cindyli - test added